app.py

```diff
@@ -29,8 +29,9 @@ def move(self, board, colour, dice_roll, make_move):
                 if move == 'end_game':
                     raise Exception()
                 try:
-                    make_move(move['location'], move['die_roll'])
-                    dice_roll.remove(move['die_roll'])
+                    rolls_moved = make_move(move['location'], move['die_roll'])
+                    for roll in rolls_moved:
+                        dice_roll.remove(roll)
```

Comment:
This is to centralise the logic for this change.

We now give `make_move` to total move we want to make, and it returns which dice_rolls were actually used to carry that out.